crates/polars-expr/src/expressions/slice.rs

```diff
@@ -118,14 +118,22 @@ impl PhysicalExpr for SliceExpr {
         let mut ac_length = results.pop().unwrap();
         let mut ac_offset = results.pop().unwrap();
 
+        // Fast path:
+        // When `input` (ac) is a LiteralValue, and both `offset` and `length` are LiteralScalar,
+        // we slice the LiteralValue and avoid calling groups() or aggregated().
+        // TODO: When `input` (ac) is a LiteralValue, and `offset` or `length` is not a LiteralScalar,
+        // we can simplify the groups calculation since we have a List containing one scalar for
+        // each group.
+
         use AggState::*;
         let groups = match (&ac_offset.state, &ac_length.state) {
             (LiteralScalar(offset), LiteralScalar(length)) => {
                 let (offset, length) = extract_args(offset, length, &self.expr)?;
 
                 if let LiteralScalar(s) = ac.agg_state() {
                     let s1 = s.slice(offset, length);
-                    ac.with_literal(s1);
+                    let s1 = s1.implode()?;
+                    ac.with_literal(s1.into_column());
```

Comment:
This seems weird to me. Why are we not aggregating here? Does this not create a scalar list literal AggregationContext?